README.md

```diff
@@ -215,6 +208,61 @@ This module has provided a simple `Record.ss` template, which assumes some basic
 If you do not use these fields, have extra fields you'd like to add, or want to change the way the fields are display,
 then you will need to override the template found under `templates/SilverStripe/Discoverer/Service/Results/Record.ss`.
 
+### Mitigating against XSS
+When handling user input (such as a search term that might be passed in on the querystring of a link) it is important
+to consider security and to understand when and where that user input needs to be sanitised.
+
+#### Sending the query to Elastic
+Generally speaking you shouldn't need to sanitise the user search term that you pass to the `Query` class. The
+Elastic client should handle this in a safe manner and do any escaping it needs to, such as escaping quotes and
+prevent the query json being manipulated.
+
+````php
+$keywords = $request->getVar($fieldKeywords);
+
+// Instantiate a new Query, and provide the search terms that we wish to search for
+$query = Query::create($keywords);
+````
+
+Also, any sanitisation that you do at this point might mean a valid search term is escaped, leading to an incorrect
+set of search results. For example, if searching for `O'Leary` you don't want to escape html entities, since
+this will convert and send `O&#039;Leary` to elastic.
+
+#### Showing the search term in the search page input field
+
+````php
+$keywords = $this->getRequest()->getVar($fieldKeywords);
+
+$fields = FieldList::create(
+    TextField::create($fieldKeywords, _t(self::class . '.FIELD_KEYWORD_LABEL', 'Search terms'), $keywords)
+        ->setInputType('search')
+);
+````
+When configuring your search form and passing the search term to a `TextField` this won't need sanitising,
+since the templating system will handle this for you. 
+
+#### Including the query on the page
+The potential for cross site-scripting (where malicious code can be inserted into the page) can occur when outputting user
+input back to the page - for example, if you wish to modify the results template to include `Showing 1 of 10 results for "my search term"`.
+
+If your implementation requires you to include the query within the results template, one way you could do this is to
+create an extension for the `Results` class and add a custom function that can be called from the Results template.
+
+```` php
+public function sanitisedQuery(Query $query): DBText
+{
+    return DBText::create()->setValue($query->getQueryString());
+}
+````
+
+In this case, because the return type is `DBText`, the Silverstripe templating system will handle safe encoding
+of `$sanitisedQuery` for you.
+
+#### Handling raw values from Elastic
+Elastic has the ability to return `raw` values on result fields. If outputting these to the template you will need
+to consider whether they are safe or whether you need to sanitise/escape the raw content. For further information,
+see https://www.elastic.co/guide/en/app-search/current/sanitization-guide.html
```

Comment:
wasn't sure about including this link since it relates to app-search

Reply:
I think its ok as it is still relevant - unless you feel like taking the relevant bits over into our guide :)